git-link.el

```diff
@@ -678,9 +678,12 @@ return (FILENAME . REVISION) otherwise nil."
 
 (defun git-link-bitbucket (hostname dirname filename _branch commit start end)
   ;; ?at=branch-name
-  (format "https://%s/%s/src/%s/%s"
+  (format "https://%s/%s/%s/%s/%s"
           hostname
           dirname
+          (if (string= "md" (downcase (or (file-name-extension filename) "")))
```

Comment:
Markdown also supports the `.markdown` extension. Given this, and the fact that the problem can occur with other markup formats, should we introduce a list, something like `bitbucket-non-source-formats` then throw `org` in there too?

Reply:
Ha, I knew you were going to suggest a list. Also interesting about org files. Here is an example org file ![image](https://github.com/user-attachments/assets/38f104d4-aaea-43e5-ad0b-f6e11659c13a) and this is the url for this: https://bitbucket.org/johnny-5-is-alive/dot-files/src/8b36a0e425ef7a002d8de0ff57a3ec81e39b075e/dev-setup-arch.org#lines-18:19

This is all to say I think for now we can support `.md` and `.markdown`. There isn't any harm in adding `org`. Let me know if I should just add `org` too.

Reply:
> This is all to say I think for now we can support .md and .markdown. There isn't any harm in adding org. Let me know if I should just add org too.

No need to add Org if they don't render it by default. 

The formats that they do render may be here: https://support.atlassian.com/bitbucket-cloud/docs/readme-content/, though unless you're looking for more to do just `.md` and `.markdown` are fine for now. 

Looking at [their docs](https://support.atlassian.com/bitbucket-cloud/docs/hyperlink-to-source-code-in-bitbucket/) it _sounds_ like the `fileviewer` query string parameter could change the view but I couldn't not find nor guess what that value would be. Similar questions [here](https://stackoverflow.com/search?q=fileviewer+%5Bbitbucket%5D). 

> @sshaw or do you just want me to reuse git-link-extensions-rendered-plain

No need. It provides for different functionality.

Reply:
> Looking at [their docs](https://support.atlassian.com/bitbucket-cloud/docs/hyperlink-to-source-code-in-bitbucket/) it sounds like the fileviewer query string parameter could change the view but I couldn't not find nor guess what that value would be. Similar questions [here](https://stackoverflow.com/search?q=fileviewer+%5Bbitbucket%5D).

Probably a good question to ask at https://community.atlassian.com/.

While we're talkin' BitBucket, their source view with that tiny `div` that stops short of the screen height and forces one to scroll is total garbage. What were they thinking!

Reply:
@sshaw I went the route you suggested.